hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/ChildFeeContextImpl.java

```diff
@@ -72,8 +74,9 @@ public Instant currentTime() {
     @Override
     public @NonNull FeeCalculator feeCalculator(@NonNull final SubType subType) {
         try {
+            var storeFactory = new ReadableStoreFactory((SavepointStackImpl) context.savepointStack());
```

Comment:
Do you think we can avoid casting to an implementation class here? How about passing the whole `context` to `createFeeCalculator()`. As an additional benefit, it would help to shorten the parameter list.

Reply:
Yes, I changed the `SavePointStackImpl` to `HederaState` interface.

In this particular case, we can replace consensus time and state with the whole context (HandleContext)
But unfortunately, this method is called also in `FeeContextImpl.feeCalculator()`, here we don’t have access to `HandleContext`. 
I also checked the classes that initialize `FeeContextImpl` - `IngestChecker` and `QueryChecker` and I don’t see a way to pass handle context from here either.